src/test/fakes.rs

```diff
@@ -291,18 +307,33 @@ impl<'a> FakeRelease<'a> {
                 fs::write(file, data)?;
             }
 
-            let prefix = format!(
-                "{}/{}/{}/{}",
-                prefix,
-                package.name,
-                package.version,
-                target.unwrap_or("")
-            );
-            log::debug!("adding directory {} from {}", prefix, path_prefix.display());
-            crate::db::add_path_into_database(&storage, &prefix, path_prefix)
+            log::debug!("adding directory {:?} from {}", kind, path_prefix.display());
+            if archive_storage {
+                let archive = match kind {
+                    FileKind::Rustdoc => rustdoc_archive_path(&package.name, &package.version),
+                    FileKind::Sources => source_archive_path(&package.name, &package.version),
+                };
+                log::debug!("store in archive: {:?}", archive);
+                let (files_list, new_alg) =
+                    crate::db::add_path_into_remote_archive(&storage, &archive, path_prefix)?;
+                let mut hm = HashSet::new();
+                hm.insert(new_alg);
+                Ok((files_list, hm))
+            } else {
+                let prefix = match kind {
+                    FileKind::Rustdoc => "rustdoc",
+                    FileKind::Sources => "sources",
+                };
+                crate::db::add_path_into_database(
+                    &storage,
+                    &format!("{}/{}/{}/", prefix, package.name, package.version,),
```

Comment:
This is missing `target`.

Reply:
the solution for both with and without archive was giving both methods the correct path, in this case `tempdir`: 

```rust
crate::db::add_path_into_database(
    &storage,
    &format!("{}/{}/{}/", prefix, package.name, package.version),
    tempdir,
)
```
and 
```rust
crate::db::add_path_into_remote_archive(&storage, &archive, tempdir)?;
```

Reply:
we have the following section in the beginning of the `upload_files` clojure:  

```rust
if let Some(target) = target {
    path_prefix.push(target);
    fs::create_dir(&path_prefix)?;
}
```

and we then put the files into `path_prefix`.